### ISSUE TITLE
Verify the URL

### ISSUE THREAD

**Issue by ctmay4** [imsweb/naaccr-geocoder-client#7]
In DMS we use the following URL:

`https://geo.naaccr.org/Services/Geocode/WebService/GeocoderWebServiceHttpNonParsedDetailed_V04_02.aspx`

But the library uses:

`https://geo.naaccr.org/Services/Geocode/WebService with query parameter to select the option.`

Then we add parameters like this:

```java
HttpUrl url = original.url()
     .newBuilder()
     .addQueryParameter("apiKey", apiKey)    // always supply the API key
     .addQueryParameter("format", "tsv")     // always parse from tsv
     .addQueryParameter("verbose", "true")   // get verbose reply
     .addQueryParameter("version", "4.02")   // set API version
     .build();
```

The unit tests verify we are getting results from the correct version, but @criderp indicated  that is should not work.

@jlorentson can you please test?

**Comment by ctmay4** [comment]
Oops.  I didn't realize the rest of the URL was in the `GeocodingService`.  @jlorentson pointed out:

> I don't know how the @GET annotation in retrofit2.http is supposed to work, but the geocode() method stub in the GeocodingService interface is annotated @GET("GeocoderWebServiceHttpNonParsed_V04_02.aspx"). I assume that is making the difference?

So I am curious if we need to set the version as a query parameter like we are doing now.  That is the only thing to check for this issue.

**Comment by criderp** [comment]
That should be changed to
`@GET("GeocoderWebServiceHttpNonParsedDetailed_V04_02.aspx")`

Also, the version parameter is required.

**Comment by ctmay4** [comment]
From @criderp 

> OK, that makes much more sense. So the library is using 
https://geo.naaccr.org/Services/Geocode/WebService/GeocoderWebServiceHttpNonParsed_V04_02.aspx
and DMS is using
https://geo.naaccr.org/Services/Geocode/WebService/GeocoderWebServiceHttpNonParsedDetailed_V04_02.aspx
The library should use the same URL as DMS.
The version parameter is required.

**Comment by jlorentson** [comment]
Yes; I just axed the line 
     .addQueryParameter("version", "4.02")   // set API version
and the test did not work. I will make the change Philip noted.

**Comment by jlorentson** [comment]
Chuck and I discussed on Friday making three versions of the main geocode() method: one each corresponding to the three URL options available from the API. These are the two mentioned in Chuck's comments above, as well as 
https://geo.naaccr.org/Services/Geocode/WebService/GeocoderWebServiceHttpNonParsedAdvanced_V04_02.aspx.
Having looked at it some more, I now think that is unnecessary. The information returned by the "plain" version is a subset of the information returned by the detailed version; any user who wants to use the "plain" version can simply ignore any fields of the GeocodeOutput object they aren't interested in. As for the advanced option, it seem that calling that with the default values for its additional parameters returns the same information as using the detailed version, so the user can pretend they are using the detailed (or plain) option and ignore the added input parameters. 

It also seems like those advanced features aren't supported yet. The updated documentation has an example with the comment 

> (XML, Verbose, All options false, with
2010 census and reference geometry) 
­ will fail to match a parcel because city is wrong 

and an example for the same address with the comment

>  (XML, Verbose, Attribute Relxation on
pre,suffix,city, with 2010 census and reference geometry) 
will succeed in matching a parcel even though city is wrong 
because the city attribute will be relaxed 

but the examples return xmls that differ only in transaction id and time taken. This is admittedly not a thorough test, but it makes me suspect those features are not yet working.

If these advanced features are indeed not yet supported (and because we don't use them), I am inclined to hold off adding the new parameters to the GeocodeInput class, lest a user think they make a difference. 

tl;dr: I suggest we switch the URL to the "advanced" version but wait to add advanced feature support for a later revision. once we are confident that they actually do something.

**Comment by ctmay4** [comment]
Sounds good.  We can always expand later.  

One other thing to consider. The "advanced" versions *may* be doing things that take longer than other versions, making the calls slower.  I'm just guessing, but there has to be some difference.

**Comment by jlorentson** [comment]
We can stick with the detailed version until we add advanced support if you're worried about performance. It's an easy fix.

**Comment by ctmay4** [comment]
@criderp wanted a specific version of the API.  It is whatever we are using in DMS now.  Please just use that one.
